### ISSUE TITLE
PytzUsageWarning: The localize method is no longer necessary

### ISSUE THREAD

**Issue by Ibo0815** [scrapinghub/dateparser#1089]
Full Usage warning: 
`PytzUsageWarning: The localize method is no longer necessary, as this time zone supports the fold attribute (PEP 495). For more details on migrating to a PEP 495-compliant implementation, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
  date_obj = stz.localize(date_obj)`

Line 35 in date_parser.py

Occurs when parsing this here: "2022-10-03 22:00:00"

Python Version: 
3.9.1

**Comment by Gallaecio** [comment]
[It should be fixed in 1.1.2, released today](https://github.com/scrapinghub/dateparser/blob/master/HISTORY.rst#112-2022-10-20). Please, see if you can reproduce the issue with 1.1.2.

**Comment by Gallaecio** [comment]
https://github.com/scrapinghub/dateparser/pull/1062#issuecomment-1293953513

**Comment by madkinsz** [comment]
👋 I can reproduce this on 1.1.4 with 
```python
import dateparser
from datetime import datetime

dateparser.parse(
    "in 3 hours",
    settings={
        "TO_TIMEZONE": "UTC",
        "RELATIVE_BASE": datetime.now(),
    },
)
```

**Comment by PythonTryHard** [comment]
I did some digging. As of writing, `dataparser`'s dependency `tzlocal`'s latest stable (since dependency is not pinned) is `4.3` which utilises `pytz-deprecation-shim`.

https://github.com/scrapinghub/dateparser/blob/7b469b1282f633a99425b60feeb4b820f7215839/dateparser/freshness_date_parser.py#L4

https://github.com/scrapinghub/dateparser/blob/7b469b1282f633a99425b60feeb4b820f7215839/dateparser/freshness_date_parser.py#L40-L41

https://github.com/scrapinghub/dateparser/blob/7b469b1282f633a99425b60feeb4b820f7215839/dateparser/freshness_date_parser.py#L74-L76

As `dateparser` eagerly `.localize()`, this leads to the deprecation warning. This has been addressed upstream [in this PR from `tzlocal`](https://github.com/regebro/tzlocal/pull/140) and is included after version `5.0b1`. Install and it should be gone, though keep in mind the risk of beta versions.

And since `tzlocal` would have completed migration away from the deprecation shim, perhaps it's time to remove the `if hasattr(obj, 'localize'):` sections, since `tzlocal` now returns `zoneinfo` objects?
